pkg/statedb2/index/int.go

```diff
@@ -0,0 +1,18 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package index
+
+import "encoding/binary"
+
+func Uint64(n uint64) Key {
+	buf := make([]byte, 8)
+	binary.BigEndian.PutUint64(buf, n)
```

Comment:
A comment explaining why big endian lends itself better to this task would be nice